transportable-udfs-plugin/src/main/java/com/linkedin/transport/plugin/TransportPlugin.java

```diff
@@ -0,0 +1,225 @@
+  }
+
+  /**
+   * Configures SourceSets, dependencies and tasks related to each Transport UDF platform
+   */
+  private void configurePlatform(Project project, PlatformConfiguration platformConfiguration,
+      SourceSet mainSourceSet, SourceSet testSourceSet) {
+    SourceSet sourceSet = configureSourceSet(project, platformConfiguration, mainSourceSet);
+    TaskProvider<GenerateWrappersTask> generateWrappersTask =
+        configureGenerateWrappersTask(project, platformConfiguration, mainSourceSet, sourceSet);
+    // TODO: shade and package into Jar
+    // Add Transport tasks to build task dependencies
+    project.getTasks().named(LifecycleBasePlugin.BUILD_TASK_NAME).configure(task -> {
+      // TODO: Replace this task with the shaded jar tasks once we create them
+      task.dependsOn(sourceSet.getClassesTaskName());
+    });
+
+    TaskProvider<Test> testTask = configureTestTask(project, platformConfiguration, mainSourceSet, testSourceSet);
+    project.getTasks().named(LifecycleBasePlugin.CHECK_TASK_NAME).configure(task -> task.dependsOn(testTask));
+  }
+
+  /**
+   * Creates and configures a {@link SourceSet} for a given platform, sets up the source directories and
+   * configurations and configures the default dependencies required for compilation and runtime of the wrapper
+   * SourceSet
+   */
+  private SourceSet configureSourceSet(Project project, PlatformConfiguration platformConfiguration,
+      SourceSet mainSourceSet) {
+    JavaPluginConvention javaConvention = project.getConvention().getPlugin(JavaPluginConvention.class);
+    Path platformBaseDir =
+        project.getBuildDir().toPath().resolve(Paths.get("generatedWrappers", platformConfiguration.getName()));
+    Path wrapperSourceOutputDir = platformBaseDir.resolve("sources");
+    Path wrapperResourceOutputDir = platformBaseDir.resolve("resources");
+
+    return javaConvention.getSourceSets().create(platformConfiguration.getName(), sourceSet -> {
+      /*
+        Creates a SourceSet and set the source directories. E.g.
+
+        presto {
+          java.srcDirs = ["${buildDir}/generatedWrappers/sources"]
+          resources.srcDirs = ["${buildDir}/generatedWrappers/resources"]
+          compileClasspath += sourceSets.main.output
+        }
+       */
+      SourceSetUtils.getSourceDirectorySet(sourceSet, platformConfiguration.getLanguage()).setSrcDirs(
+          ImmutableList.of(wrapperSourceOutputDir));
+      sourceSet.getResources().setSrcDirs(ImmutableList.of(wrapperResourceOutputDir));
+      sourceSet.setCompileClasspath(project.files(sourceSet.getCompileClasspath(), mainSourceSet.getOutput()));
+
+      /*
+        Sets up the configuration for the wrapper SourceSet. E.g.
+
+        configurations {
+          prestoImplementation.extendsFrom mainImplementation
+          prestoRuntimeOnly.extendsFrom mainRuntimeOnly
+        }
+       */
+      project.getConfigurations().getByName(sourceSet.getImplementationConfigurationName())
```

Comment:
Please use getConfigurationForSourceSet here and everywhere.

Reply:
Also, if the enum is meaningless without the sourceSet, I am still having a hard time seeing the value of the enum.

Reply:
I wouldn't say the enum is meaningless. The configuration is meaningless without the source set. But the sourceset is not available when you are defining the defaults. So we need some way to defining a mapping between the configuration type and the dependency which will be applied on to the sourceset. The enum helps us build the mapping.

Reply:
As per our discussion, I think it is okay to keep the enum as we don't have a good way to represent (SourceSet, ConfigurationType, Dependency) at compile time. We can achieve the same by removing enum and using string configuration names instead, like `prestoImplementation`, which will have weaker guarantees.

Reply:
Let us check this in. I will take a look into refactoring it afterwards. I think there is a way to make it more organized, probably by refactoring Platform, or getting rid of SourceSetUtils (by encapsulating in a non-utility class), or both.